cmd/create/cluster/cmd.go

```diff
@@ -3651,21 +3651,23 @@ func getExpectedResourceIDForAccRole(hostedCPPolicies bool, roleARN string, role
 	return strings.ToLower(fmt.Sprintf("%s-%s-Role", rolePrefix, accountRoles[roleType].Name)), rolePrefix, nil
 }
 
-func getInitialValidSubnets(aws aws.Client, ids []string, r *reporter.Object) ([]*ec2.Subnet, error) {
-	initialValidSubnets := []*ec2.Subnet{}
+func getInitialValidSubnets(aws aws.Client, ids []string, r *reporter.Object) ([]*ec2types.Subnet, error) {
+	initialValidSubnets := []*ec2types.Subnet{}
 	excludedSubnets := []string{}
 
 	validSubnets, err := aws.ListSubnets(ids...)
 
 	if err != nil {
 		return initialValidSubnets, err
 	}
+
 	for _, subnet := range validSubnets {
 		hasRHManaged := tags.Ec2ResourceHasTag(subnet.Tags, tags.RedHatManaged, strconv.FormatBool(true))
 		if !hasRHManaged {
-			initialValidSubnets = append(initialValidSubnets, subnet)
+			s := subnet
+			initialValidSubnets = append(initialValidSubnets, &s)
```

Comment:
We are converting the subnets to pointers, and then in the `create cluster` we convert them back to subnets -> `awssdk.ToString(subnet.SubnetId)`
It's better to return a slice of subnets instead.